calm/dsl/cli/mpis.py

```diff
@@ -0,0 +1,1128 @@
+
+        for entity in entity_results:
+            entity_data = entity["data"]
+            project_names = get_group_data_value(
+                entity_data, "project_names", value_list=True
+            )
+            available_to = "-"
+            if project_names:
+                project_count = len(project_names)
+                if project_count == 1:
+                    available_to = "{} Project".format(project_count)
+                else:
+                    available_to = "{} Projects".format(project_count)
+
+            categories = get_group_data_value(entity_data, "categories")
+            category = "-"
+            if categories:
+                category = categories.split(":")[1]
+
+            owner = get_group_data_value(entity_data, "owner_username")
+            if not owner:
+                owner = "-"
+
+            data_row = [
+                highlight_text(get_group_data_value(entity_data, "name")),
+                highlight_text(get_group_data_value(entity_data, "app_source")),
+                highlight_text(owner),
+                highlight_text(get_group_data_value(entity_data, "author")),
+                highlight_text(available_to),
+                highlight_text(get_group_data_value(entity_data, "version")),
+                highlight_text(category),
+                highlight_text(get_group_data_value(entity_data, "app_state")),
+                highlight_text(entity["entity_id"]),
+            ]
+
+            table.add_row(data_row)
+
+    click.echo(table)
+
+
+def get_mpi_latest_version(name, app_source=None, app_states=[]):
+
+    res = get_mpis_group_call(
+        name=name, app_states=app_states, group_member_count=1, app_source=app_source
+    )
+    group_results = res["group_results"]
+
+    if not group_results:
+        raise Exception("No MPI found with name {}".format(name))
+
+    entity_results = group_results[0]["entity_results"]
+    entity_version = get_group_data_value(entity_results[0]["data"], "version")
+
+    return entity_version
+
+
+def get_mpi_by_name_n_version(name, version, app_states=[], app_source=None):
+    """
+    It will fetch marketplace item with particular version.
+    Special case: As blueprint with state REJECTED and other can coexist with same name and version
+    """
+
+    client = get_api_client()
+    filter = "name==" + name + ";version==" + version
+
+    if app_states:
+        filter += get_states_filter(state_key="app_state", states=app_states)
+
+    if app_source:
+        filter += ";app_source=={}".format(app_source)
+
+    payload = {
+        "length": 250,
+        "filter": filter,
+    }
+    res, err = client.market_place.list(params=payload)
+    if err:
+        raise Exception("[{}] - {}".format(err["code"], err["error"]))
+
+    res = res.json()
+    if not res["entities"]:
+        message = "no mpi found with name {} and version {}.\nRun 'calm get mpis -d' to get detailed list of mpis".format(
```

Comment:
Fix this error msg.

Reply:
Add error logs instead of raising exceptions. Use `--show-trace` to show traceback. Please do it everywhere.